### PR TITLE
Add product title to header when available

### LINKED ISSUE #35172: [Header] Product name
**User story**

- As a merchant,
- I want to see what product I'm editing at all times,
- So I can avoid possible mistakes.

**Design**
5sAIeTRd9Yp7nSCT33BAWz-fi-8924%3A388910

**Acceptance criteria**

- [ ] We show the product's name as the last step in the breadcrumbs in the header.
- [ ] We use the product type if the product doesn't have a name yet. Example: `New standard product`, `New grouped product`, `New affiliate product`, `New gift card`, etc.
- [ ] The name updates dynamically when the user enters the product's name in the form field. The user doesn't need to save or update changes for the name to change.
- [ ] The character limit is 70. If the name is longer, we truncate it by adding `...` at the end of the last word. When the user hovers over it, we show the full product name in a tooltip.

### PR DESCRIPTION
**Pull request by joshuatf**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds the product title to the header when available, otherwise adds a "add new" string to the header based on product type.

Note that product types haven't been officially added yet, so a simple product type placeholder exists.

Screen Shot 2022-10-31 at 2 32 40 PM
Screen Shot 2022-10-31 at 2 28 21 PM

Closes #35172  .

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Navigate to the new product management experience ( Products -> Add new (MVP) ).
2. Note that "New product" exists in the header initially
3. Type something in the product name field
4. Note that the header is updated to reflect this name
5. Delete the content from the product name field
6. Note that the header reverts to "New product"
7. Save the product as a draft or publish
8. Remove the content from the product name field
9. Note that the previously saved name shows in the header
10. Type some content in the product name field
11. Note that the header is updated to reflect those changes

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce-admin/client/products/product-title.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+	const title = getProductTitle( values.name, values.type, persistedName );
+
+	return <WooHeaderPageTitle>{ title }</WooHeaderPageTitle>;
+};
+
+registerPlugin( 'woocommerce-product-title', {
+	render: ProductTitle,
+	icon: 'admin-generic',
+} );
```
Comment (louwie17) [inline plugins/woocommerce-admin/client/products/product-title.tsx]: This `registerPlugin` is not necessary as we are rendering it as part of the `ProductFormHeader` anyway, so we could remove this.

**Review by louwie17 — CHANGES_REQUESTED** [review]
Nice work @joshuatf, this tested well! Also thanks for separating this into different components 🎉 
I just left one comment in relation to the call to `registerPlugin` otherwise this looks good.

File: plugins/woocommerce-admin/client/products/product-title.tsx (hunk 1 above)
Comment (joshuatf) [reply plugins/woocommerce-admin/client/products/product-title.tsx]: Good catch!

**Comment by joshuatf** [comment]
Thanks for catching that @louwie17!  Pushed an update in https://github.com/woocommerce/woocommerce/pull/35431/commits/068ecab7675a21c303138d782c8f52bf819d84c9 and this should be ready for re-review.

**Review by louwie17 — DISMISSED** [review]
LGTM 🚀 Thanks for removing the `registerPlugin` logic, this looks good!

**Review by louwie17 — APPROVED** [review]
Rebase looks good, LGTM 🚀

**Pull request merged**